### PR TITLE
update some tests to run in isolated workers using tap

### PR DESCRIPTION
**Pull request by rochdev**
### What does this PR do?

Update some tests to run in isolated workers using tap.

### Motivation

We need a way to isolate tests so that we can write integration tests without complex logic to cleanup after the suite, which is also not always possible.

### CODE REVIEW

File: packages/dd-trace/test/appsec/iast/overhead-controller.spec.js [hunk 1]
```diff
@@ -402,7 +404,7 @@ describe('Overhead controller', () => {
         })
       }
 
-      testInRequest(app, tests)
+      () => testInRequest(app, tests) // TODO: re-enable when fixed with tap
```
Comment (bengl) [inline packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:407]: A bare arrow function is a pretty weird way to comment something out. Can you just use a normal comment here?

File: packages/dd-trace/test/profiling/exporters/agent.spec.js [hunk 2]
```diff
@@ -306,86 +303,88 @@ describe('exporters/agent', function () {
     })
   })
 
-  describeOnUnix('using UDS', () => {
-    let listener
-
-    beforeEach(done => {
-      url = `${path.join(os.tmpdir(), `dd-trace-profiler-test-${Date.now()}`)}.sock`
+  if (os.platform() !== 'win32') {
```
Comment (bengl) [inline packages/dd-trace/test/profiling/exporters/agent.spec.js:306]: A lot of the changes in this file look like they should be done in a separate PR. Any reason why you included them here?

**Review by bengl — CHANGES_REQUESTED** [review]

File: packages/dd-trace/test/profiling/exporters/agent.spec.js (hunk 2 above)
Comment (rochdev) [reply packages/dd-trace/test/profiling/exporters/agent.spec.js:306]: There is only 1 small change, which is replacing the `describeOnUnix` function with a condition, because `describe.skip` doesn't exist in node-tap.

File: packages/dd-trace/test/appsec/iast/overhead-controller.spec.js (hunk 1 above)
Comment (rochdev) [reply packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:407]: Sure, I was trying to avoid changing too many lines and came up with this, but I can comment out all the lines involved instead.

Comment (bengl) [reply packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:407]: Can't you just comment out this line?

Comment (rochdev) [reply packages/dd-trace/test/appsec/iast/overhead-controller.spec.js]: Fixed.

**Comment by Qard** [comment]
Lots of notes about things not working with this test runner. Is the plan to fix these before landing? Otherwise we should at least elaborate on _why_ those tests are not working so it's clear what is involved in enabling them again.

File: packages/dd-trace/test/appsec/iast/path-line.spec.js [hunk 3]
```diff
@@ -27,7 +29,8 @@ describe('path-line', function () {
   ]
 
   describe('getFirstNonDDPathAndLine', () => {
-    it('call does not fail', () => {
+    // TODO: make this work regardless of the test runner
+    it.skip('call does not fail', () => {
```
Comment (simon-id) [inline packages/dd-trace/test/appsec/iast/path-line.spec.js:33]: cc: @uurien

File: packages/dd-trace/test/exporters/scheduler.spec.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
 'use strict'
 
+require('../setup/tap')
```
Comment (simon-id) [inline packages/dd-trace/test/exporters/scheduler.spec.js:3]: This file can be deleted as it was moved under appsec folder in the RC PR and now does not exist here anymore, thus creating a merge conflict.

File: .github/workflows/appsec.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+name: AppSec
```
Comment (simon-id) [inline .github/workflows/appsec.yml:1]: Can we sync together on the appsec tests ? maybe there is a quick way to make them work with this PR instead of putting them aside.

**Review by simon-id — CHANGES_REQUESTED** [review]
Are you sure we can't find a way to include the setup script automatically for every file ? using the `NODE_ARGS` env var with a `-r` or something ?

File: .github/workflows/appsec.yml (hunk 5 above)
Comment (rochdev) [reply .github/workflows/appsec.yml:1]: Sure, although we should probably split the tests by functionalities anyway.

File: packages/dd-trace/test/appsec/iast/analyzers/command-injection-analyzer.spec.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+require('../../../../../dd-trace/test/setup/tap')
```
Comment (simon-id) [inline packages/dd-trace/test/appsec/iast/analyzers/command-injection-analyzer.spec.js:1]:
```suggestion
'use strict'

require('../../../../../dd-trace/test/setup/tap')
```

**Comment by rochdev** [comment]
> Turns out the tests were working, so we can keep the appsec test with the other ones. So is this file still relevant ?

I would say yes since it makes it easier to isolate which functionality is passing or failing.

File: .github/workflows/appsec.yml (hunk 5 above)
Comment (rochdev) [reply .github/workflows/appsec.yml:1]: I would say yes since it makes it easier to isolate which functionality is passing or failing.

**Comment by simon-id** [comment]
@uurien The CI is failing on windows for IAST tests only, please take a look.

File: packages/dd-trace/test/profiling/exporters/agent.spec.js (hunk 2 above)
Comment (simon-id) [inline packages/dd-trace/test/profiling/exporters/agent.spec.js:306]: @uurien Might be the source of the tests failing on windows only

Comment (uurien) [reply packages/dd-trace/test/profiling/exporters/agent.spec.js:306]: I don't think so, the job that fails is running this command: `npm run test:appsec -- --coverage --nyc-arg=--include=packages/dd-trace/src/appsec/**/*.js` where `"test:appsec": "tap packages/dd-trace/test/appsec/**/*.spec.js",`

So, `packages/dd-trace/test/profiling/exporters/agent.spec.js` should be ignored.

**Comment by Qard** [comment]
@simon-id I would prefer we _not_ use extra args to do that. I'd like to be able just run a single test directly more easily by just running `node path/to/test.js`.

File: packages/dd-trace/test/appsec/iast/overhead-controller.spec.js [hunk 7]
```diff
@@ -178,231 +182,231 @@ describe('Overhead controller', () => {
   })
   describe('full feature', () => {
```
Comment (tlhunter) [inline packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:183]: Why not `describe.skip`? With the test body commented out the test output sort of suggests that the test runs / the condition is covered.

Comment (rochdev) [reply packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:183]: This doesn't exist in tap.

Comment (Qard) [reply packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:183]: You _can_ do `describe(name, fn, { skip: true })`, though that may not be great on the end of a large, multi-line describe. 🤔

Comment (rochdev) [reply packages/dd-trace/test/appsec/iast/overhead-controller.spec.js:183]: Fixed.

File: packages/datadog-core/test/storage/async_hooks.spec.js [hunk 8]
```diff
@@ -1,5 +1,7 @@
 'use strict'
 
+require('../setup')
```
Comment (tlhunter) [inline packages/datadog-core/test/storage/async_hooks.spec.js:3]: Should these first two files be `require('../setup/tap')`?

**Comment by rochdev** [comment]
Superseded by #2889
